### ISSUE TITLE
Chore: replace class with hooks .

### ISSUE THREAD

**Issue by sombo20** [netojose/react-modal#35]
**I would like to remove the use of classes in the code and use only functions, to make the code easier to read and maintain**

@netojose

**Comment by netojose** [comment]
Good point @sombo20 . The core was made when hooks wasn't available yet. I agree we need to update the code.
